src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -139,15 +148,168 @@ public ObservableCollection<string> NumberFormatList
+        /// </summary>
+        public bool HideIronPythonAlertsIsChecked
+        {
+            get
+            {
+                return hideIronPAlerts;
+            }
+            set
+            {
+                hideIronPAlerts = value;
+                RaisePropertyChanged(nameof(HideIronPythonAlertsIsChecked));
+            }
+        }
+
+        /// <summary>
+        /// Controls the IsChecked property in the "Show Whitespace in Python editor" toogle button
+        /// </summary>
+        public bool ShowWhitespaceIsChecked
+        {
+            get
+            {
+                return showWhitespace;
+            }
+            set
+            {
+                showWhitespace = value;
+                RaisePropertyChanged(nameof(ShowWhitespaceIsChecked));
+            }
+        }
+
+        /// <summary>
+        /// Controls the IsChecked property in the "Node autocomplete" toogle button
+        /// </summary>
+        public bool NodeAutocompleteIsChecked
+        {
+            get
+            {
+                return nodeAutocomplete;
+            }
+            set
+            {
+                nodeAutocomplete = value;
+                RaisePropertyChanged(nameof(NodeAutocompleteIsChecked));
+            }
+        }
+
+        /// <summary>
+        /// Controls the IsChecked property in the "Enable T-spline nodes" toogle button
+        /// </summary>
+        public bool EnableTSplineIsChecked
+        {
+            get
+            {
+                return enableTSpline;
+            }
+            set
+            {
+                enableTSpline = value;
+                RaisePropertyChanged(nameof(EnableTSplineIsChecked));
+            }
+        }
+
+        /// <summary>
+        /// Gets the different Python Engine versions availables from PythonNodeModels.dll
+        /// </summary>
+        /// <returns>Strings array with the different names</returns>
+        private string[] GetPythonEngineOptions()
+        {
+            try
+            {
+                var enumType = AppDomain.CurrentDomain.GetAssemblies()
+                    .SelectMany(s =>
+                    {
+                        try
+                        {
+                            return s.GetTypes();
+                        }
+                        catch (ReflectionTypeLoadException)
+                        {
+                            return new Type[0];
+                        }
+                    }).FirstOrDefault(t => t.FullName.Equals("PythonNodeModels.PythonEngineVersion"));
+
+                return Enum.GetNames(enumType);
+            }
+            catch
+            {
+                return Array.Empty<string>();
+            }
+        }
+
+        private void AddPythonEnginesOptions()
+        {
+            var pythonEngineOptions = GetPythonEngineOptions();
+            if (pythonEngineOptions.Length != 0)
+            {
+                for (int i = 0; i < pythonEngineOptions.Length; i++)
```

Comment:
This can probably be simplified by `foreach`

Reply:
Changed it. I initially went with a "for" as it is a little bit more efficient but I understand that in this case clarity may have priority, thanks for the feedback